connection.go

```diff
@@ -279,3 +287,12 @@ func (conn *Connection) checkTimeout() error {
 		conn.Unlock()
 	}
 }
+
+func (conn *Connection) setWindowSize(n float64) {
+	if n < float64(conn.session.config.MinConnectionWindowSize) {
+		n = float64(conn.session.config.MinConnectionWindowSize)
+	}
+	conn.Lock()
```

Comment:
Remove lock here so you can use it at L277

Reply:
Yes, modified